src/main/java/org/jetbrains/kotlinx/lincheck/strategy/ManagedStrategyTransformer.java

```diff
@@ -80,58 +111,667 @@ public void visitSource(String source, String debug) {
+        if (isSynchronized) {
+            access ^= ACC_SYNCHRONIZED; // disable synchronized
+        }
+
         MethodVisitor mv = super.visitMethod(access, mname, desc, signature, exceptions);
         mv = new JSRInlinerAdapter(mv, access, mname, desc, signature, exceptions);
-        mv = new SharedVariableAccessMethodTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
-        // TODO add other transformers
+        mv = new SynchronizedBlockTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+
+        if (isSynchronized) {
+            // synchronized method is replaced with synchronized lock
+            mv = new SynchronizedBlockAddingTransformer(mname, new GeneratorAdapter(mv, access, mname, desc), className, access, classVersion);
+        }
+        mv = new UnsafeTransformer(new GeneratorAdapter(mv, access, mname, desc));
+
+        boolean shouldBeIgnored = "<clinit>".equals(mname) || shouldBeIgnored(className);
+        if (shouldBeIgnored) {
+            // static initialization sections and classes corresponding to ignoredEntryPoints
+            // should not have inserted strategy methods.
+            mv = new IgnoredSectionTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+        } else {
+            mv = new WaitNotifyTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+            mv = new ParkUnparkTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+            // SharedVariableAccessMethodTransformer should be an earlier visitor than ClassInitializationTransformer
+            // not to have suspension points before 'beforeClassInitialization' call in <clinit> block.
+            mv = new LocalObjectManagingTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+            mv = new SharedVariableAccessMethodTransformer(mname, new GeneratorAdapter(mv, access, mname, desc));
+            mv = new TimeStubTransformer(new GeneratorAdapter(mv, access, mname, desc));
+            mv = new ThreadLocalRandomTransformer(className, new GeneratorAdapter(mv, access, mname, desc));
+        }
         mv = new TryCatchBlockSorter(mv, access, mname, desc, signature, exceptions);
         return mv;
     }
 
-    class SharedVariableAccessMethodTransformer extends ManagedStrategyMethodVisitor {
-        public SharedVariableAccessMethodTransformer(String methodName, GeneratorAdapter mv) {
+    private boolean shouldBeIgnored(String className) {
+        for (String entryPoint : ignoredEntryPoints)
+            if (className.startsWith(entryPoint))
+                return true;
+        return false;
+    }
+
+    /**
+     * Changes package of transformed classes from java/util package, excluding some
+     */
+    private static class JavaUtilRemapper extends Remapper {
+        @Override
+        public String map(String name) {
+            boolean isException = name.endsWith("Exception");
```

Comment:
Let's check whether it is inherited from `Throwable` instead (seems bug-prone now)